src/components/Pagination/Pagination.js

```diff
@@ -82,11 +99,14 @@ export default class Pagination extends Component {
   handlePageInputChange = evt => {
     const page = Number(evt.target.value);
     if (
-      page > 0 &&
+      // Added props for defaultPageText and defaultItemText to be
+      // used for case when page # is 0 or empty. For other cases
+      // existing props will be used.
+      page >= 0 &&
       page <= Math.ceil(this.props.totalItems / this.state.pageSize)
     ) {
       this.setState({ page });
-      this.props.onChange({ page, pageSize: this.state.pageSize });
+      this.pageInputDebouncer(page);
```

Comment:
Should we invoke this in the callback to `setState`?

Reply:
Wrt `cWM` - Given JavaScript has no notion of "destructor", we need to pair the lifecycle callbacks for things that need clean-up. I'm guessing that `cDM`/`cWU` is the way to go.

Reply:
Added the callback.